bokehjs/src/coffee/models/canvas/cartesian_frame.coffee

```diff
@@ -42,24 +44,23 @@ export class CartesianFrame extends LayoutCanvas
         ranges[name] = extra_range
     return ranges
 
-  _get_scales: (scale_type_name, ranges, frame_range) ->
+  _get_scales: (scale, ranges, frame_range) ->
     scales = {}
+
     for name, range of ranges
-      if range.type == "Range1d" or range.type == "DataRange1d"
-        if scale_type_name == "log"
-          scale_type = LogScale
-        else
-          scale_type = LinearScale
-        range.scale_hint = scale_type_name
-      else if range.type == "FactorRange"
-        scale_type = CategoricalScale
-      else
-        logger.warn("unknown range type for range '#{name}': #{range}")
-        return null
-      scales[name] = new scale_type({
-        source_range: range
-        target_range: frame_range
-      })
+      if range instanceof DataRange1d or range instanceof Range1d
+        if not scale instanceof LogScale or not scale instanceof LinearScale
+          throw new Error("Range #{range.constructor.name} is incompatible is Scale #{scale.constructor.name}")
+      if range instanceof FactorRange
+        if not scale instanceof CategoricalScale
+          throw new Error("Range #{range.constructor.name} is incompatible is Scale #{scale.constructor.name}")
+      if scale instanceof LogScale and range instanceof DataRange1d
```

Comment:
This PR punts on the DataRange1d.scale_hint stuff. It's going to have to be addressed in it's own PR, especially if we intend to support custom scales.